src/app/loadout/mod-assignment-drawer/selectors.ts

```diff
@@ -31,9 +43,15 @@ export function useEquippedLoadoutArmor(loadout: Loadout) {
         }
       }
 
-      return loadoutDimItems.sort(compareBy((item) => bucketOrder.indexOf(item.bucket.hash)));
+      return bucketOrder
+        .map(
+          (bucketHash) =>
+            loadoutDimItems.find((item) => item.bucket.hash === bucketHash) ||
+            currentItems?.find((item) => item.bucket.hash === bucketHash)
+        )
+        .filter((item): item is DimItem => Boolean(item));
```

Comment:
I still don't think this is as easy to read as `_.compact(...)`

Reply:
Yeah my bad, I normally use compact but slipped into autopilot here (we don't have lodash at work)